controllers/managedfusion_controller.go

```diff
@@ -299,14 +299,36 @@ func (r *ManagedFusionReconciler) initReconciler(ctx context.Context, req ctrl.R
 func (r *ManagedFusionReconciler) reconcilePhases() (reconcile.Result, error) {
 	if !r.managedFusionSecret.DeletionTimestamp.IsZero() {
 		r.Log.Info("Managed Fusion configuration Secret is marked for deletion, checking for ManagedFusionOfferings CRs on the cluster")
-		if found, err := r.checkForManagedOfferings(); err != nil {
-			return ctrl.Result{}, err
-		} else if found {
-			r.Log.Info("ManagedFusionOffering CRs found, issuing delete")
-			offering := v1alpha1.ManagedFusionOffering{}
-			if err := r.deleteAllOf(&offering); err != nil {
-				return ctrl.Result{}, fmt.Errorf("failed to issue delete for one or more ManagedFusionOffering CRs, %v", err)
-			} else {
+		offeringList := v1alpha1.ManagedFusionOfferingList{}
+		if err := r.list(&offeringList); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to list ManagedFusionOffering CRs, %v", err)
+		}
+		if len(offeringList.Items) > 0 {
+			var found bool
+			var errors []error
+			for i := range offeringList.Items {
+				offering := offeringList.Items[i]
```

Comment:
This will create a copy, please use a pointer instead.
```suggestion
				offering := &offeringList.Items[i]
```